packages/mui-base/src/utils/useTransitionStatus.ts

```diff
@@ -19,6 +19,9 @@ export function useTransitionStatus(open: boolean, enabled = true) {
   if (enabled) {
     if (open && !mounted) {
       setMounted(true);
+      if (transitionStatus !== 'entering') {
+        setTransitionStatus('entering');
+      }
```

Comment:
This is necessary after the change, and also seems to fix the CSS Transition Group demo in the popups experiment. Likely should have been the case originally.